newIDE/app/src/MainFrame/index.js

```diff
@@ -1522,597 +1498,554 @@ class MainFrame extends React.Component<Props, State> {
-            !!this.props.unsavedChanges &&
-            this.props.unsavedChanges.hasUnsavedChanges
+      </Drawer>
+      <Toolbar
+        ref={toolbar}
+        showProjectIcons={!props.integratedEditor}
+        hasProject={!!currentProject}
+        toggleProjectManager={toggleProjectManager}
+        exportProject={() => openExportDialog(true)}
+        requestUpdate={props.requestUpdate}
+        simulateUpdateDownloaded={simulateUpdateDownloaded}
+        simulateUpdateAvailable={simulateUpdateAvailable}
+      />
+      <ClosableTabs hideLabels={!!props.integratedEditor}>
+        {getEditors(state.editorTabs).map((editorTab, id) => {
+          const isCurrentTab = getCurrentTabIndex(state.editorTabs) === id;
+          return (
+            <ClosableTab
+              label={editorTab.label}
+              key={editorTab.key}
+              active={isCurrentTab}
+              onClick={() => _onChangeEditorTab(id)}
+              onClose={() => _onCloseEditorTab(editorTab)}
+              onCloseOthers={() => _onCloseOtherEditorTabs(editorTab)}
+              onCloseAll={_onCloseAllEditorTabs}
+              onActivated={() => _onEditorTabActive(editorTab)}
+              closable={editorTab.closable}
+            />
+          );
+        })}
+      </ClosableTabs>
+      {getEditors(state.editorTabs).map((editorTab, id) => {
+        const isCurrentTab = getCurrentTabIndex(state.editorTabs) === id;
+        return (
+          <TabContentContainer key={editorTab.key} active={isCurrentTab}>
+            <ErrorBoundary>
+              {editorTab.render(isCurrentTab, state)}
+            </ErrorBoundary>
+          </TabContentContainer>
+        );
+      })}
+      <LoaderModal show={showLoader} />
+      <HelpFinder
+        open={helpFinderDialogOpen}
+        onClose={() => openHelpFinderDialog(false)}
+      />
+      <Snackbar
+        open={state.snackMessageOpen}
+        autoHideDuration={3000}
+        onClose={_closeSnackMessage}
+        ContentProps={{
+          'aria-describedby': 'snackbar-message',
+        }}
+        message={<span id="snackbar-message">{state.snackMessage}</span>}
+      />
+      {!!renderExportDialog &&
+        state.exportDialogOpen &&
+        renderExportDialog({
+          onClose: () => openExportDialog(false),
+          onChangeSubscription: () => {
+            openExportDialog(false);
+            openSubscription(true);
+          },
+          project: state.currentProject,
+        })}
+      {!!renderCreateDialog &&
+        state.createDialogOpen &&
+        renderCreateDialog({
+          open: state.createDialogOpen,
+          onClose: () => openCreateDialog(false),
+          onOpen: (storageProvider, fileMetadata) => {
+            openCreateDialog(false);
+            // eslint-disable-next-line
+            useStorageProvider(storageProvider)
+              .then(() => openFromFileMetadata(fileMetadata))
+              .then(state => openSceneOrProjectManager(state));
```

Comment:
This is not working for web-app at all, while everything seems to work pretty well for electron :/ I'm calling `openFileFromMetadata` and `openSceneOrProjectManager` in a similar way they are getting called from `chooseProject`. I can't really understand what's going wrong.

Reply:
Are you getting an error message? Can you post it here?
It might be because if you're trying Google Drive, it's not working locally.

Reply:
There is no error, the create dialog just closes, and nothing happens. I'm tracking every render and the state of that render, so the `fileMetadata` and `currentProject` is set correctly, but immediately on the next render everything falls back to `null`

Reply:
> It might be because if you're trying Google Drive, it's not working locally.

Yes, that doesn't work too, but this is not the issue here, I'm unable to open even the examples, there is some issue involving state here.

Reply:
Alright I found what's wrong, babel transpiles async functions to generators for browser, and native promises for node. This is why it is running without any issue on electron but not on browsers 😅 I'm converting everything back to promises.

Reply:
Great investigation and great finding!
Using promises everywhere is fine :)